atomic_reactor/plugins/pre_pull_base_image.py

```diff
@@ -54,7 +60,32 @@ def __init__(self, tasker, workflow, parent_registry=None, parent_registry_insec
             self.parent_registry = None
             self.parent_registry_insecure = False
 
-    def resolve_base_image(self, build_json):
+    def run(self):
+        """
+        Pull parent images and retag them uniquely for this build.
+        """
+        build_json = get_build_json()
+        base_image_str = str(self.workflow.builder.base_image)
+        nonce = 0
```

Comment:
Consider using `enumerate` instead.

Reply:
Thanks, makes sense.

    for nonce, parent in enumerate(sorted(self.workflow.builder.parent_images.keys())):